internal/painter/software/draw_test.go

```diff
@@ -65,8 +86,9 @@ func TestDrawImage_Contain(t *testing.T) {
 
 func TestDrawImage_ContainX(t *testing.T) {
 	target := image.NewNRGBA(image.Rect(0, 0, 100, 50))
-	img := canvas.NewImageFromImage(makeTestImage())
+	img := canvas.NewImageFromImage(makeTestImage(3, 5))
```

Comment:
Are these values the wrong way round? I assumed that when you created the rectangular pattern it was intended to be in the same orientation as the contains test?

Reply:
not, I just want a not 1:1 aspect ratio

Reply:
I guess I don't understand why this changed from before. The new parameters were added for this, and the following, function - so I was curious why the change.

Reply:
Still, I'm not sure that we cover all possibilities... maybe I'll add more test cases.